### PR TITLE
Add initializeMethodRunAddressForSnapshot function

### PR DESCRIPTION
**Pull request by ThanHenderson**
Signed-off-by: Nathan Henderson <nathan.henderson@ibm.com>

### CODE REVIEW

**Comment by ThanHenderson** [comment]
@babsingh Could you review? Related to the revert here: https://github.com/eclipse-openj9/openj9/pull/20763/

File: runtime/oti/vm_api.h [hunk 1]
```diff
@@ -1274,7 +1273,27 @@ initializeMethodRunAddress(J9VMThread *vmThread, J9Method *method);
...
+ * reinitialized to a state that is restore friendly (i.e. set to approprite interpreter entries).
+ * initializeMethodRunAddress should not be used; it may run the J9HOOK_VM_INITIALIZE_SEND_TARGET
+ * hook that initializes invocation counts and sets J9Method::methodRunAddress's to JIT specific
+ * send targets. initializeMethodRunAddressNoHook should not be used because it does not perform
+ * MethodHandle and VarHandle send target initialization, nor does it reset J9Method::extra
+ * appropriately.
+ *
+ * @param vm pointer to the J9JavaVM
+ * @param method pointer to the J9Method
+ * @return void
```
Comment (babsingh) [inline runtime/oti/vm_api.h:1292]: `@return` is not needed for `void`. just remove this instance. i will create another PR to have it fixed in the whole file. 
```suggestion
```

**Comment by babsingh** [comment]
Jenkins test sanity.functional alinux jdk21

**Comment by babsingh** [comment]
Jenkins compile win jdk11

**Comment by ThanHenderson** [comment]
I noticed a spelling mistake in one of my comments. I'll push again once the tests are done.

**Comment by babsingh** [comment]
All Windows machines are offline, so the Windows PR build will fail. The alinux PR build has passed, which is sufficient to test these changes. Feel free to push the fix for the typo whenever you're ready.

Old PR builds:
- JDK21 alinux sanity.functional: https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/6706/
- JDK11 Windows compile: https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/6707/

**Review by babsingh — APPROVED** [review]

**Pull request merged**